src/Types/BigIntType.php

```diff
@@ -28,12 +31,23 @@ public function getBindingType(): ParameterType
     /**
      * @param T $value
      *
-     * @return (T is null ? null : string)
+     * @return (T is null ? null : int|string)
      *
      * @template T
      */
-    public function convertToPHPValue(mixed $value, AbstractPlatform $platform): ?string
+    public function convertToPHPValue(mixed $value, AbstractPlatform $platform): int|string|null
     {
-        return $value === null ? null : (string) $value;
+        if ($value === null) {
+            return null;
+        }
```

Comment:
```suggestion
        if ($value === null || is_int($value)) {
            return $value;
        }
```

Let's exit here already if we already hold an integer in our hand. This prevents accidental int-to-string casts. This should also help us getting the tests to pass.